python/lsst/qa/explorer/consolidateQATable.py

```diff
@@ -0,0 +1,56 @@
+"""Command-line task and associated config for consolidating QA tables.
+
+
+The deepCoadd_qa table is a table with QA columns of interest computed
+for all filters for which the deepCoadd_obj tables are written. 
+"""
+from lsst.daf.persistence.butler import Butler
+from lsst.pex.config import (Config, Field, ConfigField, ListField, DictField, ConfigDictField,
+                             ConfigurableField)
+from lsst.pipe.base import Task, CmdLineTask, ArgumentParser, TaskRunner, TaskError
+from lsst.coadd.utils import TractDataIdContainer
+from lsst.pipe.tasks.multiBand import MergeSourcesTask, MergeSourcesConfig
+from lsst.pipe.tasks.multiBand import _makeGetSchemaCatalogs
+from lsst.coadd.utils.coaddDataIdContainer import ExistingCoaddDataIdContainer
+from lsst.coadd.utils import TractDataIdContainer
+
+from collections import defaultdict
+import functools
+import re, os
+import pandas as pd
+
+from .parquetTable import ParquetTable
+from .tractQADataIdContainer import TractQADataIdContainer
+
+# Question: is there a way that LSST packages store data files?
+ROOT = os.path.abspath(os.path.dirname(__file__))
+
+class ConsolidateQATableConfig(Config):
+    coaddName = Field(dtype=str, default="deep", doc="Name of coadd")
+
+class ConsolidateQATableTask(CmdLineTask):
+    """Write filter-merged source tables to parquet
```

Comment:
This Task does patch-merging, not filter-merging, right?

Docstring aside, I'd be a bit more comfortable if this Task and the `deepCoadd_qa_tract` dataset didn't have to exist, as it's *just* duplication of other tables we're already storing in Parquet format.  I'd instead prefer to give `ParquetTable` the ability to read (lazily, as always) from multiple per-patch Parquet files on-the-fly.  That's tricky, of course, because it means the tract-level `ParquetTable` would need to be initialized from multiple `butler.get` calls.  There may be a way to do it with a single `butler.get` call by using composite dataset functionality (possibly in Gen2 Butler as well as Gen3), but I think the first step would be to add an API like:
```
     class ParquetTable:

        @staticmethod
        def fromTract(butler, datasetType, tract, patches):
            """Return a single ParquetTable that mimics a full-tract DataFrame by
            consolidating patch-level ParquetTables on-the fly.
            """
            result = butler.get(datasetType, tract=tract, patches[0])
            for patch in patches[1:]:
                result.concatenateInPlace(butler.get(datasetType, tract=tract, patch=patch)
            return result
            
        ...
```
Would you anticipate any performance problems with an approach like this?

Thinking further ahead, if something like this (or a composite-based single-`get` version) becomes the main one people use, that'd also give us the freedom to change the implementation of `ParquetTable` to one that uses an internal Butler without breaking code that uses these objects.  That could be desirable if we want to put the actual details of the I/O behind the Butler again so we can consider using other column-access-friendly storage formats for the same datasets.  I suspect that's how we'll want to go in Gen3 eventually; `ParquetTable` would become a lazy-loading convenience interface that *uses* a Butler, rather than a lazy-loading interface that is retrieved *from* a Butler.

I don't want to push for all of those changes now - I'd rather focus on getting something up and running.  But I would like to make the changes that would let us make those changes later without breaking (much) downstream code.

And, on that note, we may want to consider renaming `ParquetTable` to something like `LazyDataFrame` and making sure its APIs are appropriate for any format we could lazy-load into a DataFrame, not just Parquet.

Reply:
My desire to have tract-level tables comes mostly from looking forward to wanting the interactive QA plots to be nearly instantaneous, even when operating on ~tens of tracts simultaneously.  For this use case, my instinct is that your suggested approach---81 butler get calls per tract, then concatenating them into a single pandas dataframe---will be significantly slower than a single parquet -> pandas load per tract.  I can run some tests, but I'm pretty sure that the snappiness you gain from running these few extra tens of seconds or whatever it is during processing will be worth the extra few MB of storage.